src/wp-includes/comment.php

```diff
@@ -2429,9 +2429,17 @@ function wp_new_comment_notify_postauthor( $comment_id ) {
 
 	$maybe_notify = $is_note ? get_option( 'wp_notes_notify' ) : get_option( 'comments_notify' );
 
+	// By default, only notify for approved comments and notes.
+	if (
+		! isset( $comment->comment_approved ) ||
+		( '1' !== $comment->comment_approved && ! $is_note ) ) {
+			$maybe_notify = false;
+	}
```

Comment:
Maybe this refactor into a single `if/elseif/else` statement is easier to reason about?

```suggestion
	// By default, only notify for approved comments and notes.
	if (
		! isset( $comment->comment_approved ) ||
		( '1' !== $comment->comment_approved && ! $is_note ) 
	) {
		$maybe_notify = false;
	} else if ( $is_note ) {
		$maybe_notify = get_option( 'wp_notes_notify' );
	} else {
		$maybe_notify = get_option( 'comments_notify' )
	}
```

Reply:
Aside: Will `$comment->comment_approved` ever be not-set or `null`? It's default value is `'1'` and it is typed as a `string`. Doesn't hurt, but just curious. If so, then `! isset( $comment->comment_approved )` could be removed.

Reply:
This will change the logic slightly. If get_option( 'comments_notify' ) is true and '1' !== $comment->comment_approved and the type is a comment, the default should be false. This would make it true. I will try to make the logic clearer, maybe remove the trinary or add inline docs.

Reply:
I may have just provided a poor suggestion, so don't feel like it needs to be actioned on 😊